### PR TITLE
Customizable installation logic

### PR DESCRIPTION
**Pull request by rlp-apple**
Separates out installation logic so that it can easily be overridden in
the event of local changes (for example: adding certs to generated
packages).

### CODE REVIEW

File: cmake/install_directives.cmake [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+        "tools/deployment/osquery.example.conf"
+
+      DESTINATION
+        "/private/var/osquery"
+    )
+
+    install(
+      FILES
+        "tools/deployment/macos_packaging/pkg/com.facebook.osqueryd.conf"
+        "tools/deployment/macos_packaging/pkg/com.facebook.osqueryd.plist"
```
Comment (sharvilshah) [inline cmake/install_directives.cmake:297]: Not sure where these came from, with the latest release of osquery 5.0.1, `com.facebook...` identifiers are no longer in use on macOS, instead they are now renamed to `io.osquery.agent.conf` and `io.osquery.agent.plist`. There are a couple of macOS specific files missing as well.

Can you please make sure that this is rebased with all the latest changes?

**Review by sharvilshah — CHANGES_REQUESTED** [review]

File: cmake/install_directives.cmake [hunk 2]
```diff
@@ -0,0 +1,312 @@
+function(generateInstallDirectives)
```
Comment (sharvilshah) [inline cmake/install_directives.cmake:1]: Missing copyright headers

File: cmake/install_directives.cmake (hunk 1 above)
Comment (rlp-apple) [reply cmake/install_directives.cmake:297]: When I originally put this together we had just finished updating to 4.9.0 :D  Clearly I am too slow.  I'll get it done.

File: cmake/install_directives.cmake (hunk 2 above)
Comment (rlp-apple) [reply cmake/install_directives.cmake:8]: Fixed.

File: cmake/install_directives.cmake (hunk 1 above)
Comment (rlp-apple) [reply cmake/install_directives.cmake]: OK, try now please.

**Comment by sharvilshah** [comment]
This looks good, thanks @rlp-apple!

**Review by sharvilshah — APPROVED** [review]

**Review by directionless — APPROVED** [review]

**Pull request merged**